File: problems/3001/Solution.cs
public class Solution {
    private bool Between(int x, int y, int z) => System.Math.Min(x, z) < y && y < System.Math.Max(x, z);

    public int MinMovesToCaptureTheQueen(int a, int b, int c, int d, int e, int f) {
        bool rookBlocked = (a == e && c == a && Between(b, d, f)) ||
                           (b == f && d == b && Between(a, c, e));
        if ((a == e || b == f) && !rookBlocked) return 1;

        bool bishopAttacks = System.Math.Abs(c - e) == System.Math.Abs(d - f);
        bool bishopBlocked = bishopAttacks && System.Math.Abs(a - e) == System.Math.Abs(b - f) &&
                             Between(c, a, e) && Between(d, b, f);
        return bishopAttacks && !bishopBlocked ? 1 : 2;
    }
}

File: problems/3001/Solution.java
class Solution {
    private boolean between(int x, int y, int z) {
        return Math.min(x, z) < y && y < Math.max(x, z);
    }

    public int minMovesToCaptureTheQueen(int a, int b, int c, int d, int e, int f) {
        boolean rookBlocked = (a == e && c == a && between(b, d, f)) ||
                              (b == f && d == b && between(a, c, e));
        if ((a == e || b == f) && !rookBlocked) return 1;

        boolean bishopAttacks = Math.abs(c - e) == Math.abs(d - f);
        boolean bishopBlocked = bishopAttacks && Math.abs(a - e) == Math.abs(b - f) &&
                                between(c, a, e) && between(d, b, f);
        return bishopAttacks && !bishopBlocked ? 1 : 2;
    }
}
